datasets/me/acp_peps/crawler.py

```diff
@@ -0,0 +1,162 @@
+
+
+def extract_latest_filing(
+    dates: List[Dict[str, object]]
+) -> Tuple[Optional[str], Optional[int]]:
+    if not dates:
+        return None, None
+
+    # Sort the dates in descending order
+    dates_sorted = sorted(
+        dates,
+        key=lambda item: datetime.strptime(item["datum"], "%Y-%m-%d"),
+        reverse=True,
+    )
+
+    # Assume the first entry is the latest
+    latest_entry = dates_sorted[0]
+    latest_date: Optional[str] = latest_entry.get("datum")
+    latest_report_id: Optional[int] = latest_entry.get("stariIzvjestaj")
+
+    # Check if the latest report is -1, move to the next valid entry
+    if latest_report_id == -1:
+        for entry in dates_sorted[1:]:
+            if entry.get("stariIzvjestaj") != -1:
+                latest_date = entry.get("datum")
+                latest_report_id = entry.get("stariIzvjestaj")
+                break
+        else:
+            # If all entries have -1, set the value to None
+            latest_report_id = None
+
+    return latest_date, latest_report_id
+
+
+def fetch_and_process_csv(context: Context, latest_report_id: int):
+    # URL for fetching the CSV
+    url = f"https://portal.antikorupcija.me:9343/acamPublic/izvestajDetailsCSV.json?izvestajId={latest_report_id}"
+
+    # Fetch the ZIP file containing the CSV
+    zip_path = context.fetch_resource(f"{latest_report_id}.zip", url)
+    try:
+        with ZipFile(zip_path, "r") as zip:
+            # Find the file name that matches the pattern
+            file_name = next(
+                (
+                    name
+                    for name in zip.namelist()
+                    if name.startswith("csv_funkcije_funkcionera_")
+                ),
+                None,
+            )
+            # Check if the file was found
+            if not file_name:
+                context.log.warning("No matching file found in the ZIP archive.")
+                return []
+
+            with zip.open(file_name) as zfh:
+                fh = io.TextIOWrapper(zfh, encoding="utf-8-sig")
+                reader = DictReader(fh, delimiter=",", quotechar='"')
+                return list(reader)
+    except BadZipFile:
+        context.log.warning(f"Failed to open {zip_path} as a ZIP file. Skipping.")
+        return []
+
+
+def crawl_person(context: Context, person):
+    name = person.pop("imeIPrezime")
+    # position = person.pop("nazivFunkcije")
+    dates = person.pop("izvjestajImovine")
+    latest_date, latest_report_id = extract_latest_filing(dates)
+    report_details = fetch_and_process_csv(context, latest_report_id)
+    if not report_details:
+        return
+
+    for row in report_details:
+        person_name = row.pop("FUNKCIONER_IME")
+        person_surname = row.pop("FUNKCIONER_PREZIME")
+        function = row.pop("FUNKCIJA")
+        # organization = row.pop("ORGANIZACIJA")
+        start_date = row.pop("DATUM_POCETKA_OBAVLJANJA", None)
+        end_date = row.pop("DATUM_PRESTANKA_OBAVLJNJA")
```

Comment:
so this just leaves the last row's function assigned to function, right? What I usually do is make an empty set, then in a loop over positions, add to the set if it passes make_position, categorise and make_occupancy https://github.com/opensanctions/opensanctions/blob/me-income-and-asset-reports/datasets/hr/public_officials/crawler.py#L148

and then emit the person (and the position entities) if there were any position entities in the set.